x/observer/abci.go

```diff
@@ -0,0 +1,36 @@
+package observer
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/zeta-chain/zetacore/x/observer/keeper"
+	"github.com/zeta-chain/zetacore/x/observer/types"
+	"math"
+)
+
+func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
+	lastBlockObserverCount, found := k.GetLastObserverCount(ctx)
+	if !found {
+		ctx.Logger().Error("LastBlockObserverCount not found at height", ctx.BlockHeight())
+		return
+	}
+
+	allObservers := k.GetAllObserverMappers(ctx)
+	totalObserverCountCurrentBlock := 0
+	for _, observer := range allObservers {
+		totalObserverCountCurrentBlock += len(observer.ObserverList)
+	}
+	if totalObserverCountCurrentBlock < 0 {
+		ctx.Logger().Error("TotalObserverCount is negative at height", ctx.BlockHeight())
+		return
+	}
+	if totalObserverCountCurrentBlock == int(lastBlockObserverCount.Count) {
+		return
+	}
+	ctx.Logger().Error("LastBlockObserverCount does not match the number of observers found at current height", ctx.BlockHeight())
```

Comment:
Does this error mean that something bad happened in the last block? Is this an unexpected error and where/how are we handling this error for now? btw, was just curious

Reply:
Yeah, this line of code would only be executed when the observer set changes. The change in the validator set can be triggered from one of these hooks for example 
https://github.com/zeta-chain/zeta-node/blob/7eadeaf4339b570b9e983872cab7e7d50aa4eb37/x/observer/keeper/hooks.go#L10-L12

We are just logging the error , and pausing the inbound for now

Reply:
Thanks. I thought it's the code for long-term.